### PR TITLE
Workflow schema validation to the validate script

### PR DESCRIPTION
**Pull request by aparna-ravindra**
The current "validate" script that runs on PR and Push validates the .properties.json files. This PR adds an additional validation to check the schema of the .yaml file.

### CODE REVIEW

File: ci/npm-publish.yml [hunk 1]
```diff
@@ -47,4 +47,4 @@ jobs:
       - run: npm ci
       - run: npm publish
         env:
-          NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+          NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (ashwinsangem) [inline ci/npm-publish.yml:50]: if there's no difference between the above lines, you can skip this file from the PR.

File: ci/python-publish.yml [hunk 2]
```diff
@@ -33,4 +33,4 @@ jobs:
       uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
       with:
         user: __token__
-        password: ${{ secrets.PYPI_API_TOKEN }}
+        password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (ashwinsangem) [inline ci/python-publish.yml:36]: same as above.

File: script/validate-data/index.ts [hunk 3]
```diff
@@ -65,30 +64,48 @@ async function checkWorkflows(folders: string[]): Promise<WorkflowWithErrors[]>
...
+
+    const workflowFileContent = await fs.readFile(workflowPath, "utf8");
+    const workflow = safeLoad(workflowFileContent); // Validate yaml parses without error
+    
+    let workflowValidator = new validator();
+    const workflowSchema = require("./workflow-schema.json");
+    
+    const workflowValidationResult = workflowValidator.validate(workflow, workflowSchema)
+    const workflowValidationErrors = workflowValidationResult.errors.map(e => e.toString())
+    workflowErrors.errors = workflowErrors.errors.concat(workflowValidationErrors)
```
Comment (bishal-pdMSFT) [inline script/validate-data/index.ts:107]: these two statements can be clubbed together.

File: script/validate-data/index.ts [hunk 4]
```diff
@@ -65,30 +64,48 @@ async function checkWorkflows(folders: string[]): Promise<WorkflowWithErrors[]>
...
+  }
+
+  try {
+    workflowErrors.errors = await validateWorkflowProperties(propertiesPath)
+
+    const workflowFileContent = await fs.readFile(workflowPath, "utf8");
+    const workflow = safeLoad(workflowFileContent); // Validate yaml parses without error
+    
+    let workflowValidator = new validator();
+    const workflowSchema = require("./workflow-schema.json");
```
Comment (bishal-pdMSFT) [inline script/validate-data/index.ts:103]: How will this schema file be maintained and updated? If the schema goes out of date with what in production, then we will get false negatives. Hence it is very important to define the clear path for maintenance of this file.

cc @andymckay

**Pull request closed without merging**